src/Application/Asset/TerminalAssetPersistenceService.php

```
@@ -0,0 +1,116 @@
+    public function __construct(
+        private readonly AssetRepositoryInterface $assets,
+        private readonly AssetTerminalStatusCacheInterface $assetTerminalStatusCache,
+    ) {
+    }
+
+    public function findAsset(AssetId $assetId): ?Asset
+    {
+        try {
+            return $this->assets->findById($assetId);
+        } catch (\Throwable $exception) {
+            throw RepositoryUnavailableException::forReason(self::REPOSITORY_FAILURE_REASON, $exception);
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\App\Domain\Asset\Exception\RepositoryUnavailableException' in method 'findAsset'. (undefined)

(StaticAccess)